### ISSUE TITLE
A11y_VirtualAgentBots_Topics_EditConversationTopic_Keyboard: Static element such as text and disabled controls receive keyboard focus.

### ISSUE THREAD

**Issue by MayuriNaik** [OfficeDev/office-ui-fabric-react#10715]
**Environment Details:**
Browser Version: Microsoft Edge 44.18362.267.0
Windows Version: Windows10
Screen reader: Narrator, NVDA (2019.2)
MAS Violated: MAS 2.1.1

**Steps to Reproduce:**
1. Home page should open for Virtual Agent Bots.
2. Navigate to Hamburger Menu at the left of the screen and select "Topics". 
3. From the List of topics under "Topics" heading, select any topic by clicking its "Edit Conversation" button. Screen with node chart to edit conversation would appear.
4. Navigate to the Menu bar and try to navigate to the "Store Hours" static element and disabled controls.

**Actual:**
Issue 1:
When keyboard user navigates through the Menu bar on the screen the static elements such as text i.e. "Store Hours" receive keyboard focus but on activating it no action takes place.

Issue 2:
When keyboard user navigates through the Menu bar, the controls such as "Discard changes", "Undo", etc. which visually appear as disabled but still receive keyboard focus which is incorrect as the users will consider it as enabled controls and will try to interact with it.

**Expected:**
Issue 1:
The "Store hour" state element should not receive keyboard focus while navigating through the screen.

Issue 2:
The controls which visually appear as disabled should not receive keyboard focus. These controls should be available in the focus order of the keyboard users only when they are in the enabled state.

**User Impact:**
Issue 1:
If state element which is available on the screen receive keyboard focus then the keyboard users will misunderstand the control as an interactive element and try to activate it.

Issue 2:
If the disabled controls receive keyboard focus then wrong information will be conveyed to the keyboard users that these control as in the enabled state and this might lead to confusion.

**Comment by khmakoto** [comment]
Hi @MayuriNaik , I think what you're logging is an application bug that needs to be investigated by someone from the Virtual Agent Bots team since it is not clear if this is indeed a Fabric bug. Please contact that team first and, if it is indeed a Fabric issue then we can start working on it. Thanks!

**Comment by rkalivarapu** [comment]
> Hi @MayuriNaik , I think what you're logging is an application bug that needs to be investigated by someone from the Virtual Agent Bots team since it is not clear if this is indeed a Fabric bug. Please contact that team first and, if it is indeed a Fabric issue then we can start working on it. Thanks!

This is an issue with Office fabric disabled command bar buttons. Please find the attachment for the same issue where disabled buttons gets keyboard focus.
https://developer.microsoft.com/en-us/fabric#/controls/web/commandbar

![fabric_commandbar_disabled](https://user-images.githubusercontent.com/22643928/66233073-4991ef00-e69f-11e9-9f2f-dab82eb9b9df.gif)

**Comment by rkalivarapu** [comment]
> Thanks for taking the time to enter an issue. However, it seems that there aren't enough details here for this issue to be actionable.
> When issues are created, we need details such as:
> 
> Which Fabric component and which version of the Fabric component is causing the issue
> What behaviors and attributes are missing or incorrect
> What you expected and what is actually happening
> Confirmation that the problem reproduces in isolation
> 
> Without a clear understanding of these details, it's not possible to take clear action on issues. We are unable to meet your expectations, properly address the root cause, and make changes without affecting the expectations of other consumers.
> Please provide these additional details as you are able. The default issue template provides an outline of these details and is viewable when creating a new issue. Additionally, if this is an accessibility issue, please see Accessibility Troubleshooting in our wiki for more guidance. If these details cannot be provided, please kindly close the bug.
> Thank you for your patience.

This is an issue with Office fabric disabled command bar buttons. Please find the attachment for the same issue where disabled buttons gets keyboard focus.
https://developer.microsoft.com/en-us/fabric#/controls/web/commandbar

![fabric_commandbar_disabled](https://user-images.githubusercontent.com/22643928/66233073-4991ef00-e69f-11e9-9f2f-dab82eb9b9df.gif)

**Comment by khmakoto** [comment]
@MayuriNaik @rkalivarapu I've verified that this reproduces in the `CommandBar` on our website. However, I think this is by design because our we allow for focus on disabled elements on purpose.

If you don't want this behavior though, there is an option for you to disallow this focus. This is done by passing `allowDisabledFocus: false` as a prop in the `CommandBarItems`. I've exemplified this in [the following codepen](https://codepen.io/khmakoto/pen/GRRJLzB?editors=0010) where I make the `Download` button not focusable via keyboard navigation.

Hope this helps and feel free to contact me if you need anything else!

**Comment by rkalivarapu** [comment]
Thanks for the update and support. It would be great if it is updated in the documentation https://developer.microsoft.com/en-us/fabric#/controls/web/commandbar
